test/node/byte_utils.test.ts

```diff
@@ -0,0 +1,479 @@
+    }
+  }
+];
+const fromBase64Tests: ByteUtilTest<'fromBase64'>[] = [
+  {
+    name: 'should create buffer from base64 input',
+    inputs: [Buffer.from('abc\u{1f913}', 'utf8').toString('base64')],
+    expectation({ web, output, error }) {
+      if (!isNode16OrLater && web) {
+        // Skip reason: btoa and atob were not made globals until node 16
+        expect(error).to.be.instanceOf(ReferenceError);
+        return;
+      }
+      expect(error).to.be.null;
+      expect(output).to.deep.equal(Buffer.from('abc\u{1f913}', 'utf8'));
+    }
+  },
+  {
+    name: 'should return empty buffer for empty string input',
+    inputs: [''],
+    expectation({ web, output, error }) {
+      if (!isNode16OrLater && web) {
+        // Skip reason: btoa and atob were not made globals until node 16
+        expect(error).to.be.instanceOf(ReferenceError);
+        return;
+      }
+      expect(output).to.have.property('byteLength', 0);
+    }
+  }
+];
+const toBase64Tests: ByteUtilTest<'toBase64'>[] = [
+  {
+    name: 'should create base64 string from buffer input',
+    inputs: [Buffer.from('abc\u{1f913}', 'utf8')],
+    expectation({ web, output, error }) {
+      if (!isNode16OrLater && web) {
+        // Skip reason: btoa and atob were not made globals until node 16
+        expect(error).to.be.instanceOf(ReferenceError);
+        return;
+      }
+      expect(error).to.be.null;
+      expect(output).to.deep.equal(Buffer.from('abc\u{1f913}', 'utf8').toString('base64'));
+    }
+  },
+  {
+    name: 'should return empty string for empty buffer input',
+    inputs: [Buffer.alloc(0)],
+    expectation({ web, output, error }) {
+      if (!isNode16OrLater && web) {
+        // Skip reason: btoa and atob were not made globals until node 16
+        expect(error).to.be.instanceOf(ReferenceError);
+        return;
+      }
+      expect(error).to.be.null;
+      expect(output).to.be.a('string').with.lengthOf(0);
+    }
+  }
+];
+const fromHexTests: ByteUtilTest<'fromHex'>[] = [
+  {
+    name: 'should create buffer from hex input',
+    inputs: [Buffer.from('abc\u{1f913}', 'utf8').toString('hex')],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.deep.equal(Buffer.from('abc\u{1f913}', 'utf8'));
+    }
+  },
+  {
+    name: 'should return empty buffer for empty string input',
+    inputs: [''],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.have.property('byteLength', 0);
+    }
+  },
+  {
+    name: 'should use leading valid hex characters',
+    inputs: ['abxxcd'],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.deep.equal(Buffer.from('abxxcd', 'hex'));
+    }
+  },
+  {
+    name: 'should slice input strings down to nearest even length (len: 1)',
+    inputs: ['a'],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.deep.equal(Buffer.from('a', 'hex'));
```

Comment:
it's not clear how the test is testing the "slice down to nearest even length" since it's using the result of `Buffer.from` in the assertion on the exact same string; and I have a similar comment for the case below that, too